src/test/java/edu/kis/powp/jobs2d/events/ICompoundCommandListener.java

```diff
@@ -0,0 +1,39 @@
+package edu.kis.powp.jobs2d.events;
+
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.ICompoundCommandInstance;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import edu.kis.powp.jobs2d.drivers.DriverManager;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ICompoundCommandListener implements ActionListener {
+
+    private DriverManager driverManager;
+
+    public ICompoundCommandListener(DriverManager driverManager) {
+        this.driverManager = driverManager;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        List<DriverCommand> commands = new ArrayList<DriverCommand>();
+        commands.add(new OperateToCommand(0, -100));
+        commands.add(new OperateToCommand(30, -100));
```

Comment:
Testing DEEP copy requires a compound command embedded in another compound command (with modification of the embedded original).